snuba/clickhouse/http.py

```diff
@@ -31,20 +31,21 @@ def __repr__(self) -> str:
 
 
 class HTTPBatchWriter(BatchWriter):
-    def __init__(self, schema, host, port, options=None, table_name=None):
+    def __init__(self,
+        schema,
+        host,
+        port,
+        encoder: Callable[[Iterable[WriterTableRow]], Iterable[bytes]],
```

Comment:
I'm totally in favor of moving the encoder here, but I think it might make a bit more sense to have the encoder signature be `Callable[WriterTableRow, bytes]`. With that, I think it would be a bit better to make `chunked` an `Optional[int]` that when `None` means "don't chunk" (essentially the old writer behavior, before streaming) and if an `int`, would be the chunk size (`1` would maintain the current behavior), something larger would be somewhere in between what you have here and what we have now, which probably makes more sense for the current consumer-driven writer? I think it makes more sense for the writer to control this process than the encoder in general.